allensdk/test/brain_observatory/behavior/test_stimulus_processing.py

```diff
@@ -64,13 +148,118 @@ def test_get_draw_epochs(start_frame, stop_frame, expected):
+                                              'gratings_180.0',
+                                              'gratings_270.0', 'omitted'],
+                               'image_category': ['grating', 'grating',
+                                                  'grating', 'grating',
+                                                  'omitted'],
+                               'image_set': ['grating', 'grating', 'grating',
+                                             'grating', 'omitted'],
+                               'phase': [10, 10, 10, 10, None],
+                               'frequency': [90, 90, 90,
+                                                     90, None]}),
+                             ({}, ['images', 'grating'],
+                              {'image_index': [0],
+                               'image_name': ['omitted'],
+                               'image_category': ['omitted'],
+                               'image_set': ['omitted'],
+                               'phase': [None],
+                               'frequency': [None]})],
+                         indirect=['behavior_stimuli_data_fixture'])
+def test_get_stimulus_metadata(behavior_stimuli_data_fixture,
+                               remove_stimuli, expected_metadata):
+    for key in remove_stimuli:
+        # do this because at current images are not tested and there's a
+        # hard coded path that prevents testing when this is fixed this can
+        # be removed.
+        del behavior_stimuli_data_fixture['items']['behavior']['stimuli'][key]
+    stimulus_metadata = get_stimulus_metadata(behavior_stimuli_data_fixture)
 
+    expected_df = pd.DataFrame.from_dict(expected_metadata)
+    expected_df.set_index(['image_index'], inplace=True, drop=True)
 
-# def test_get_visual_stimuli_df():
-#     pass
-#     # TODO
-#     # See above (this is a dependency of get_stimulus_presentations)
+    assert stimulus_metadata.equals(expected_df)
+
+
+@pytest.mark.parametrize("behavior_stimuli_time_fixture,"
+                         "behavior_stimuli_data_fixture, "
+                         "expected", [
+                             ({"timestamp_count": 15, "time_step": 1},
+                              {"images_set_log": [
+                                  ('Image', 'im065', 5, 0),
+                                  ('Image', 'im064', 25, 6)
+                              ],
+                                  "images_draw_log": (([0] * 2 + [1] * 2 +
+                                                       [0] * 3) * 2 + [0]),
+                                  "grating_set_log": [
+                                      ("Ori", 90, 3.5, 0),
+                                      ("Ori", 270, 15, 6)
+                                  ],
+                                  "grating_draw_log": (([0] + [1] * 3 + [0] * 3)
+                                                       * 2 + [0])},
+                              {"duration": [3.0, 2.0, 3.0, 2.0],
```

Comment:
I'm a little confused about this test case. So the gratings are null for `image_name` field? Where's the "frequency" and "phase"?

Reply:
When these are loaded from the set_log and merged in the get_stimulus_presentations function the name for gratings is set to None and the orientation for images is set to None, this is why the output is formed as such. I didn't touch the logic of this function except adding to the assertion of expected size so this is a design decision of the previous implementation.

Reply:
Additionally this function does not load any metadata about either images or gratings it only loads the Image name, orientation and the data to define when the stimuli occurred. The metadata is handled (which includes phase and frequency in my implementation) only when calling the get_stimulus_metadata function. This was done to line up to the previous implementation.